src/model_config_tests/test_config.py

```diff
@@ -223,3 +185,97 @@ def test_license_file(self, control_path):
         assert content == license, (
             f"LICENSE file should be equal to {LICENSE} found here: " + LICENSE_URL
         )
+
+
+def read_exe_manifest_fullpaths(control_path: Path):
+    """Return the full paths to the executables in the executable manifest file"""
+    manifest_path = control_path / "manifests" / "exe.yaml"
+    with open(manifest_path) as f:
+        _, data = yaml.safe_load_all(f)
+    exe_fullpaths = {item["fullpath"] for item in data.values()}
+    return exe_fullpaths
+
+
+def read_config_model_exes(config: dict[str, Any]):
+    """Return the exe values of the model and sub-model defined in config.yaml"""
+    exes = []
+    if "exe" in config:
+        exes.append(config["exe"])
+    for model in config.get("submodels", []):
+        if "exe" in model:
+            exes.append(model["exe"])
+    return exes
+
+
+def check_manifest_exes_in_spack_location(
+    model_module_name, model_repo_name, control_path, config
+):
+    """This compares executable paths in the executable manifest, and checks
+    they match an install path in the spack.location release artefact. The
+    version defined in the module configuration in config.yaml, is used
+    to find the relevant release version.
+
+    This is called in model-specific config tests.
+
+    Parameters
+    ----------
+    model_module_name: str
+        Expected module name in the config.yaml file. This is used to find the version of the model
+    model_repo_name: str
+        Name of the ACCESS-NRI model repository. This is used to retrieve released spack.location
+    control_path: Path
+        The path to configuration directory
+    config: Dict[str, Any]
+        The contents of the config.yaml file
+    """
+    help_msg = (
+        "Expected module for the model is added to loaded modules in config.yaml. "
+        "The module also requires a released version. E.g.\n"
+        "   modules:\n"
+        "     use:\n"
+        f"       - {RELEASE_MODULE_LOCATION}\n"
+        "     load:\n"
+        f"       - {model_module_name}/<version>\n"
+        "Model executable paths can then be filenames that found in paths added by loaded module"
+    )
+
+    # Check module is defined in configuration file
+    assert "modules" in config and "load" in config["modules"], help_msg
+    loaded_modules = config["modules"]["load"]
+    modules = [m for m in loaded_modules if m.startswith(f"{model_module_name}/")]
+    assert len(modules) == 1, help_msg
+
+    # Extract out the version
+    _, module_version = modules[0].split("/")
+
+    # Use the module version to download spack.location file
+    url = (
+        f"https://github.com/ACCESS-NRI/{model_repo_name}/"
+        f"releases/download/{module_version}/spack.location"
+    )
```

Comment:
Ah, now I understand why we don't do it for Prereleases - you're getting it from the repo not from the environment. If you did want to get it from the the environment, you would find the `spack.location` file in `/g/data/vk83(/prerelease)?/apps/spack/<version>/spack/var/spack/environments/<model>_<version>/spack.location`.

Reply:
Ok, just thinking on how to get it from environment, we would need to know the spack version (e.g. `0.20`, `0.22`). The models used in configs might use different spack versions, so this might not be able to be a fixed Github variable.

If module use paths, were the the longer paths in `config.yaml, we could infer more information e.g.
```
modules:
 use:
    - /g/data/vk83/apps/spack/0.20/release/modules/linux-rocky8-x86_64
 load:
    - access-om2/2024.03.0
```
The test code could be able refer the spack versions from `0.20`, and that it's not `prerelease` with `/g/data/vk83/apps` rather than `/g/data/vk83/prerelease/apps`. The `<model>_<version>` from `access-om2/2024.03.0` will need to converted to `access-om2-2024_03_0` to build path: 
```
/g/data/vk83/apps/spack/0.20/spack/var/spack/environments/access-om2-2024_03_0/spack.location
```
But then this can the con of using `/g/data/vk83/apps/spack/0.20/release/modules/linux-rocky8-x86_64` over `/g/data/vk83/modules/access-models`. But I don't have any other ideas on how to otherwise find the spack version.. 

I know there's been some talk about opening PR in `<model>-configs` when there's a new pre-release model built so repro tests can be run. I am wondering for these PRs with pre-release models that only `repro` tests  are run, and that `config`/QA tests are for PRs that will be merged into `release-*` branches.  For example, say if there was a workflow that would  open a PR and dynamically configure `config.yaml` to use the prerelease module path and module version. It would/should fail the `config` tests, because the exe manifest has not been updated (it needs a `payu setup` on gadi to be run to update the manifest files). It also should fail some tests, because we want to ensure there's a released `spack.location`, e.g. we don't want to have `prerelease` module in released configurations.

Yeah I am not really too sure, what do you think?

Reply:
Oh wait... QA tests aren't run on gadi, so I wouldn't be able read `/g/data/vk83/apps/spack/0.20/spack/var/spack/environments/access-om2-2024_03_0/spack.location` anyway

Reply:
Shelving this bit for now, I think.